include/tdi/common/tdi_table_key.hpp

```diff
@@ -34,6 +34,30 @@ namespace tdi {
 // Forward declarations
 class Table;
 
+/**
+ * @brief Key Field Type. A key can have multiple fields, each with a different
+ * type
+ */
+enum class KeyFieldType {
+  INVALID = 0,
+  EXACT = 1,
+  TERNARY = 2,
+  RANGE = 3,
+  LPM = 4,
+  OPTIONAL = 5,
+};
```

Comment:
Please change this. This already exists under tdi_defs.h. The way it needs to be structured is an enum split across Core, arch and target